### ISSUE TITLE
Synced Rendering Capability on JupyterLab

### ISSUE THREAD

**Issue by aladinoster** [executablebooks/jupyterlab-myst#144]
## Proposal

I find interesting the sync capabilities between views and editors in [jupyter lab](https://jupyterlab.readthedocs.io/en/latest/user/file_formats.html#markdown) (details on [video here](https://youtu.be/eQsRlqK-z1c)). I was wondering if it is possible to have the same capabilities enhanced with MyST. Is this somehow already supported?

<img width="1777" alt="image" src="https://github.com/executablebooks/jupyterlab-myst/assets/9197388/e30d4a37-58c9-4a9d-8914-0e40f3cfaf07">

Thanks!

## Additional notes

Tested on: 
```
jupyterlab: 3.6.3
jupyterlab_myst: 1.2.0
```

**Comment by rowanc1** [comment]
There is a really good start at this that just merged in #102. A few notes on what I have noticed:

- [x] images are not being mapped correctly #145
- [ ] the frontmatter is being removed I think by Jupyter

In this example, the full yaml block is removed before we get to the renderer, which means we can't show the title etc. @agoose77 any idea how to get that to stop being removed (I think by Jupyter?!).

![image](https://github.com/executablebooks/jupyterlab-myst/assets/913249/eb631e45-ef05-4281-bcb0-07fa1da0f3e0)

**Comment by agoose77** [comment]
@rowanc1 i hadn't noticed the images in my tests!

Jupyterlab has a setting for the markdown viewer that is off by default - I initially thought this was broken too!

**Comment by rowanc1** [comment]
Nice - that does the trick! It is annoying that the setting truncates the model before it gets to the renderer.

Do you know if we can access those settings and show a warning? A very hacky way is: But I am sure there is some Jupyter way of doing it.

```typescript
    console.log('Using jupyterlab-myst:mimeRenderer');
    console.log(app);
    setTimeout(() => {
      const hideFrontMatter = (app as any)?._pluginMap?.[
        '@jupyterlab/markdownviewer-extension:plugin'
      ]?.service?.currentWidget?.content?._config?.hideFrontMatter;
      console.log(hideFrontMatter);
    }, 3000);
```

**Comment by agoose77** [comment]
> It is annoying that the setting truncates the model before it gets to the renderer.

I agree. It's a prime example of where there's too many assumptions being made in the JLab frontend w.r.t Markdown rendering.

> Do you know if we can access those settings and show a warning? 

Yes, we can listen to the settings for the appropriate plugin:

https://github.com/jupyterlab/jupyterlab/blob/d98ef15bfa916309c82d105a5d442d915d5dd1df/packages/markdownviewer-extension/src/index.ts#L100-L111

**Comment by rowanc1** [comment]
Yep happy to setup a project -- @stevejpurves and I were trying to get a bit of an order of operations here:

- #141

Maybe that could be moved into a milestone and add a bit of organization to that. A small update that I would like to add is the `exercise` directives, which @mmcky and a few others are looking for. Maybe that can be the feature cutoff for a 1.x series and then we get the JLab 4.0 branch over the line? The only other piece we could do is citations?

**Comment by rowanc1** [comment]
We have released these changes in v2.0.0, which is designed for JupyterLab 4. Looking forward to your feedback @aladinoster!
